src/Workleap.Extensions.OpenApi.Analyzers/CompareTypedResultWithAnnotationAnalyzer.cs

```diff
@@ -0,0 +1,279 @@
+﻿using Microsoft.CodeAnalysis;
+using System.Collections.Immutable;
+using Microsoft.CodeAnalysis.Diagnostics;
+using Workleap.Extensions.OpenApi.Analyzers.Internals;
+
+namespace Workleap.Extensions.OpenAPI.Analyzers;
+
+[DiagnosticAnalyzer(LanguageNames.CSharp)]
+public class CompareTypedResultWithAnnotationAnalyzer : DiagnosticAnalyzer
+{
+    private const string DiagnosticId = "WLOAS001";
+    
+    internal static readonly DiagnosticDescriptor AnnotationMustMatchTypedResult = new(
+        id: DiagnosticId,
+        title: "Mismatch between annotation return type and endpoint return type",
+        messageFormat: "Mismatch between annotation return type and endpoint return type",
+        category: "OpenAPI",
+        DiagnosticSeverity.Error,
+        isEnabledByDefault: true);
+
+    public override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics => ImmutableArray.Create(AnnotationMustMatchTypedResult);
+
+    public override void Initialize(AnalysisContext context)
+    {
+        context.EnableConcurrentExecution();
+        context.ConfigureGeneratedCodeAnalysis(GeneratedCodeAnalysisFlags.None);
+
+        context.RegisterCompilationStartAction(compilationContext =>
+        {
+            var analyzerContext = new AnalyzerContext(compilationContext.Compilation);
```

Comment:
You should add something like

````c#
if (!analyzerContext.IsValid)
    return;
````

The idea is to avoid doing work is a type, such as ProducesResponseSymbol, is null. This allows reducing CPU usage in the context where the analyzer won't have anything to do.

Reply:
Added a check inside the Rule where if there are no attributes on symbol, then skip. Is that what you had in mind?
```
if (context.Symbol.GetAttributes().Length == 0)
            {
                return;
            }
```

Reply:
`new AnalyzerContext(compilationContext.Compilation);` try to find some types using `compilation.GetTypeByMetadataName`. If the types are not found, there is no reason to call `RegisterSymbolOperation` as you won't be able to provide any diagnostic.

Reply:
got it, added a isValid check for IResult and ProducesResponseType symbol